BDArmory/Weapons/Missiles/BDModularGuidance.cs

```diff
@@ -1007,7 +1005,7 @@ private void CheckMiss(Vector3 targetPosition)
             if (MissileState != MissileStates.PostThrust) return;
             if (GuidanceMode == GuidanceModes.Orbital) return; 
             // if I'm to close to my vessel avoid explosion
-            if ((vessel.CoM - SourceVessel.CoM).magnitude < 4 * DetonationDistance) return;
+            if ((vessel.CoM - SourceVessel.CoM).sqrMagnitude < 16 * DetonationDistance) return;
```

Comment:
This should have an extra `DetonationDistance`, but I notice that `DetonationDistance` is using a special value of `-1` for some things in MissileBase.cs. It might be worth adding
```csharp
public float DetonationDistanceSqr => DetonationDistance > 0 ? DetonationDistance * DetonationDistance : -1;
```
on line 91 of MissileBase.cs and using that everywhere `DetonationDistance * DetonationDistance` is currently being used.